### PR TITLE
[google_workspace] Add support for Alert Data Stream

### LINKED ISSUE #1725: Google Workspace Alerts Center
### Description
Google Workspace (formerly G Suite) Alert Center provides extensive visibility into threats detected in Workspace tenants. Admins can manage alerts more efficiently through the unified view that the alert center provides. Additionally, it provides insights that help them assess their organization’s exposure to internal and external security issues at the domain and user levels.

The alert center provides several out of the box alerts relating to activities such as phishing, user management, authentication and device management. Integrating these alerts into Elastic SIEM would provide analysts with the ability to correlate alerts from Workspace with other environmental data and signals. It would serve as a complimentary module to our existing Workspace integration, which is based on the Reports API.

### Architecture
The [Alert Center API ](https://developers.google.com/admin-sdk/alertcenter/guides)can be used to ingest alerts and supporting information.

# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists
- [ ] Sample event (`sample_event.json`) exists

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Added a data stream.
- Added data collection logic to the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files
- Added dashboards and visualizations.
- Added test for pipeline for the data stream.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.4.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/google_workspace directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes #1725
- Related https://github.com/elastic/beats/pull/33400

## Screenshots

image
image
image
image
image
image

### CODE REVIEW

File: packages/google_workspace/data_stream/alert/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+        - yyyy-MM-dd'T'HH:mm:ss.SSSZ
+        - yyyy/MM/dd HH:mm:ss z
+      target_field: google_workspace.alert.end_time
+      on_failure:
+        - append:
+            field: error.message
+            value: '{{{_ingest.on_failure_message}}}'
+  - set:
+      field: event.end
+      copy_from: google_workspace.alert.end_time
```
[the remark this answers is not in the document]
Comment (vinit-elastic) [reply packages/google_workspace/data_stream/alert/elasticsearch/ingest_pipeline/default.yml:82]: Hey @marc-gr - We think that the end users who are new to elastic are more familiar with the product-specific field names and might not be completely aware of all the ECS mappings. Therefore, we thought of giving users an option to have custom mappings, as they have field names similar to the original platform. Moreover, the User can remove duplicates in order to save storage simply by toggling a button once he gets hold of the package and its mapping.

File: packages/google_workspace/data_stream/alert/agent/stream/httpjson.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+{{#if http_client_timeout}}
+request.timeout: {{http_client_timeout}}
+{{/if}}
+request.transforms:
+  - set:
+      target: url.params.pageSize
+      value: {{page_size}}
+  - set:
+      target: url.params.filter
+      value: '[[sprintf "createTime >= %q" (formatDate (parseDate .cursor.last_create_time))]]'
```
[the remark this answers is not in the document]
Comment (vinit-elastic) [reply packages/google_workspace/data_stream/alert/agent/stream/httpjson.yml.hbs:25]: Hey @marc-gr - I have looked at the data collection of the Alert Data stream. However, when I used Filebeat to do the same I found a strange behavior ( at least it looked strange to me ). I saw that in the initial request, we were receiving Page = 0. Moreover, in the subsequent pagination requests, the page count increased as intended. 
However, from the second interval, in the initial request, we had page = 1.  And for all the pages the count was increased by 1 in all the pages. 

Below are the tables comparing values of `page` key.

First Request:
Page No | Expected Page | Actual Page
-- | -- | --
Request 1 | 0 | 0
page 1 | 1 | 1
page 2 | 2 | 2
page 3 | 3 | 3

Second Request
Page No | Expected Page | Actual Page
-- | -- | --
Request 2 | 0 | 1
page 1 | 1 | 2
page 2 | 2 | 3
page 3 | 3 | 4

Let me know if my understanding is wrong.

[the remark this answers is not in the document]
Comment (andrewkroh) [reply packages/google_workspace/data_stream/alert/agent/stream/httpjson.yml.hbs:25]: Let's grab the [`request.tracer.filename`](https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-input-httpjson.html#_request_tracer_filename) log (and redact any tokens). Then we can see exactly what's going out in the requests and coming back in the responses.

[the remark this answers is not in the document]
Comment (marc-gr) [reply packages/google_workspace/data_stream/alert/agent/stream/httpjson.yml.hbs:25]: > Let's grab the [`request.tracer.filename`](https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-input-httpjson.html#_request_tracer_filename) log (and redact any tokens). Then we can see exactly what's going out in the requests and coming back in the responses.

👍  This is not expected behaviour so if we can reproduce the issue might mean there is some problem with the underlying input.

**Comment by jamiehynds** [comment]
@andrewkroh @marc-gr can you confirm where we're at with this integration? It goes back to July, so hoping to get the review complete or get any changes required by Crest done ASAP. Thanks!

**Comment by marc-gr** [comment]
> @andrewkroh @marc-gr can you confirm where we're at with this integration? It goes back to July, so hoping to get the review complete or get any changes required by Crest done ASAP. Thanks!

last week we merged the required fix for the pagination to work, just waiting for a confirmation that it works for them and overall that would be it

**Comment by andrewkroh** [comment]
From talking with @vinit-elastic there is one more issue found while testing the latest snapshot that affects the page number state. They will open a new issue for elastic/beats to track the problem and link it to the related integrations PRs.

**Review by marc-gr — APPROVED** [review]

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**